server/render/index.js

```diff
@@ -127,3 +127,17 @@ export function serverRender( req, res ) {
 		res.render( 'index.jade', context );
 	}
 }
+
+export function serverRenderError( err, req, res, next ) {
+	if ( err ) {
+		if ( config( 'env' ) !== 'production' ) {
+			console.error( err );
+		}
+		req.error = err;
+		res.status( err.status || 500 );
+		res.render( '500.jade', req.context );
```

Comment:
This causes `/theme/gsfadgn` to render the 500 template instead of the 404 one :slightly_frowning_face: